docs/tokenModels/bestPractice.md

```diff
@@ -0,0 +1,16 @@
+# Best Practice
```

Comment:
A few initial things to think about:

- only store as `literal`s data that is refered to in a current process guardrail or that there is an exceedingly high probability of including in a guardrail in the future. The only other time a literal should be used is if there is a clear requirement that a value be available at all times
- all tokens should have `type` and `state` literals which should be the prodominant way in which we guard workflows. If the problem set requires a very clearly heirarchical state machine it is acceptable to have additional "sub-state" literals
- assigned roles should consistently describe the persona of an actor with respect to that token type. This means an account may be a `supplier` for one purchase order but a `buyer` for another. Also this largely means guardrails should always be in place to require that roles are copied over from one token to the next equivelent state.
- The splitting up of data across multiple metadata files should be based on how visibility should be assigned. In other words only split metadata into multiple files if there is a use case where only some of the information in that file should be released. This will become important when we start encrypting file contents and verifiably sharing data
- When a token array (for example an item/row in a purchase order) is associated with a container token (in this case a purchase order) a mechanism for requiring these match up should be implemented. This will likely be in the form of a `TokenId` metadata item on the item/row token associated with the initial PO token. (do we have a restriction for this currently?)

Reply:
Thank you, all great points, I'll add to the best practice
An alternative to `sub-state` literals could be using `or` on `state` in guard rails? 
Restrictions for tokens that are items/rows of a container - what we did before was have a restriction on the PO to have `TokenId` metadata with keys `recipe0`, `recipe1` ... `recipeN`. I think the reasoning was that a `recipe` could be used by multiple `PO`s so linking on the `recipe` may be difficult? Maybe it depends on if it's a one-to-many, many-to-many relationship

Reply:
first bullet point answered in review - in cases where many states share same hierachy, use sub state, otherwise `or` is fine.